watertap/property_models/IX_prop_pack.py

```diff
@@ -0,0 +1,1110 @@
+                    )
+                if var[ind].is_fixed():
+                    flags[(k, v_name, ind)] = True
+                    if value(var[ind]) != val:
+                        raise ConfigurationError(
+                            "While using the calculate_state method on {sb_name}, {v_name} was "
+                            "fixed to a value {val}, but it was already fixed to value {val_2}. "
+                            "Unfix the variable before calling the calculate_state "
+                            "method or update var_args."
+                            "".format(
+                                sb_name=sb.name,
+                                v_name=var.name,
+                                val=val,
+                                val_2=value(var[ind]),
+                            )
+                        )
+                else:
+                    flags[(k, v_name, ind)] = False
+                    var[ind].fix(val)
+
+            if degrees_of_freedom(sb) != 0:
+                raise RuntimeError(
+                    "While using the calculate_state method on {sb_name}, the degrees "
+                    "of freedom were {dof}, but 0 is required. Check var_args and ensure "
+                    "the correct fixed variables are provided."
+                    "".format(sb_name=sb.name, dof=degrees_of_freedom(sb))
+                )
+
+        # Solve
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            results = solve_indexed_blocks(opt, [self], tee=slc.tee)
+            solve_log.info_high(
+                "Calculate state: {}.".format(idaeslog.condition(results))
+            )
+
+        if not check_optimal_termination(results):
+            _log.warning(
+                "While using the calculate_state method on {sb_name}, the solver failed "
+                "to converge to an optimal solution. This suggests that the user provided "
+                "infeasible inputs, or that the model is poorly scaled, poorly initialized, "
+                "or degenerate."
+            )
+
+        # unfix all variables fixed with var_args
+        for (k, v_name, ind), previously_fixed in flags.items():
+            if not previously_fixed:
+                var = getattr(self[k], v_name)
+                var[ind].unfix()
+
+        # fix state variables if hold_state
+        if hold_state:
+            fix_state_vars(self)
+
+        return results
+
+
+@declare_process_block_class("IXStateBlock", block_class=_IXStateBlock)
+class IXStateBlockData(StateBlockData):
+    def build(self):
+        """Callable method for Block construction."""
+        super(IXStateBlockData, self).build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        # Add state variables
+
+        self.flow_mol_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize=0.1,
+            bounds=(1e-8, None),
```

Comment:
```suggestion
            bounds=(0, None),
```

We are moving away from using these small numbers as the bounds and using 0 instead.

Reply:
Good to know, I will edit the prop pack accordingly.

Reply:
@kurbansitterley note that if you are transitioning to using the ion property model and eliminating this extra prop model, you can merge main into your branch since a recent PR that was merged included setting all lower bounds (that were previously set to very small numbers) to 0.